### PR TITLE
outfly: init at v0.14.0

### PR DESCRIPTION
**Pull request by 71rd**
## Description of changes

Outfly is an open source 3d  game allowing the player to fly around the rings of jupiter.
Available at:
https://codeberg.org/outfly/outfly
https://yunicode.itch.io/outfly

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ou/outfly/package.nix
Comment (zimward) [inline pkgs/by-name/ou/outfly/package.nix:75]:
```suggestion
cargoHash = "sha256-EbzMPS2udrqVJF3Y2NfNV71Mv9oW8ZepQoP08w62QuY=";
```
See #327063

File: maintainers/maintainer-list.nix
Comment (zimward) [inline maintainers/maintainer-list.nix:250]:
```suggestion
    github = "71rd";
```

File: pkgs/by-name/ou/outfly/package.nix
Comment (zimward) [inline pkgs/by-name/ou/outfly/package.nix:71]: is far as i can tell this is unnecessary. (and fails when using `cargoHash`)

File: pkgs/by-name/ou/outfly/package.nix
Comment (zimward) [inline pkgs/by-name/ou/outfly/package.nix:91]: the code is licensed under GPL-3 but some assets are licensed under CC-BY-4.0 and should probably be added.

**Review by zimward — COMMENTED** [review]
Your init commit is signed but your maintainer isn't (and there is no key mentioned) did you forget to disable signing?

File: pkgs/by-name/ou/outfly/package.nix
Comment (hut) [reply pkgs/by-name/ou/outfly/package.nix:91]: Assets are not just under CC-BY-4.0, but also:

- CC-BY-3.0
- CC-BY-SA-3.0
- [Pixabay Content License](https://pixabay.com/service/license-summary)
- SIL OPEN FONT LICENSE Version 1.1
- CC0 and public domain (not sure if this is relevant)
- [A custom license](https://bjj.mmedia.is/acknow.html), stating "[the assets] are publicly available. You do not need a special permission to use them"

File: pkgs/by-name/ou/outfly/package.nix
Comment (71rd) [reply pkgs/by-name/ou/outfly/package.nix]: thank you, that is an artifact from an earlier version thatsome repos from outside crates.io. 
and nixpkgs rust vendor scripts behaved incompatibly to cargo-vendor.

File: pkgs/by-name/ou/outfly/package.nix
Comment (71rd) [reply pkgs/by-name/ou/outfly/package.nix]: @zimward thank you for the quick review!

have been thinking about this and im not sure what the correct approach is here.

The assumption here was that the entire game and its source could be treated as GPL-3. All assets appear to be under licenses that should allow for adaptations/derivative work to be licensed in a GPL-3 compatible manner.

If the game and its source can not be be considered a single work for this purpose, or this violates the intent of the meta.license attribute im not sure what to do about the Pixabay license.
It would require a custom license and it seems to have a number of restrictions on unmodified works, including forbidding selling it, making the package at probably at least unfree.
But im happy to add licenses for the assets if that is the correct way

Unsurprisingly im not a lawyer and my knowledge of copyright and how nixpkgs approaches licenses is limited.
If you can point at any guidance on how to use licenses in nixpkgs that would be appreciated.

Comment (zimward) [reply pkgs/by-name/ou/outfly/package.nix]: generally all licenses should be added as they are. if the Pixabay license makes it `unfree` ,`unfreeRedistributeable` or `free` is not something i can answer either. i asked on the nixos matrix channel, if no one answers in the next few days maybe ping emily.

Comment (71rd) [reply pkgs/by-name/ou/outfly/package.nix]: i will keep an eye on the matrix channel, and otherwise follow up directly with the author of the repo about use and licensing of assets, thanks

Comment (emilazy) [reply pkgs/by-name/ou/outfly/package.nix]: The [Pixabay licence](https://pixabay.com/service/license-summary/), for reference. The two iffy parts seem to be:

> You cannot sell or distribute Content (either in digital or physical form) on a Standalone basis. Standalone means where no creative effort has been applied to the Content and it remains in substantially the same form as it exists on our website.

IIRC there is some font licence or similar that also prohibits verbatim redistribution that is considered FOSS because you can just make some trivial modification or bundle it with something else. So that might not be an issue. On the other hand:

> You cannot use Content in any immoral or illegal way, especially Content which features recognisable people.
> You cannot use Content in a misleading or deceptive way.
> You cannot use any of the Content as part of a trade-mark, design-mark, trade-name, business name or service mark.

Some of these use restrictions seem potentially problematic (e.g. “immoral” has shades of the infamous JSON “Don’t Be Evil” clause). So I’d say that the conservative thing to do would be to include the other normal licences and then a `lib.licenses.unfree` (or `lib.licenses.unfreeRedistributable`, but usually that implies permission for verbatim redistribution) for this Pixabay one.

It’s a pretty unfortunate licence but maybe upstream would be willing to replace those assets?

You could also separate out the assets into a different package to ensure that Hydra can still build the actual engine.

Comment (71rd) [reply pkgs/by-name/ou/outfly/package.nix]: Thanks for weighing in. after talking to the author i will wait for the next release, which should resolve the licensing problematic

Comment (hut) [reply pkgs/by-name/ou/outfly/package.nix]: In OutFly [v0.11.0](https://codeberg.org/outfly/outfly/releases/tag/v0.11.0), all non-free assets have been replaced by free ones.

Comment (71rd) [reply pkgs/by-name/ou/outfly/package.nix]: thanks for the update and resolving this so quickly!
i will try to finish this after rust 1.81 makes it into nixpkgs master

**Comment by 71rd** [comment]
The current version runs and has replaced assets with unclear licensing, so lets try again

**Review by emilazy — DISMISSED** [review]
Thank you for working with upstream to resolve the licensing concerns, that’s really great! Broadly LGTM now; some concerns in the comments.

File: pkgs/by-name/ou/outfly/package.nix
Comment (71rd) [reply pkgs/by-name/ou/outfly/package.nix:5]: oh yes, that was just a remnant from using this outside nixpkgs

File: pkgs/by-name/ou/outfly/package.nix
Comment (71rd) [reply pkgs/by-name/ou/outfly/package.nix:9]: those were just left in after i tested support for those explicitly, but there is no reason we "need" them here.

File: pkgs/by-name/ou/outfly/package.nix
Comment (71rd) [reply pkgs/by-name/ou/outfly/package.nix:66]: Unfortunately the paths are stripped in fixup. iirc cargo opens external dependencies generally using dlopen, and since that is not detected by fixup, those paths are marked as unnecessary

File: pkgs/by-name/ou/outfly/package.nix
Comment (71rd) [reply pkgs/by-name/ou/outfly/package.nix:94]: There is theoretical support for macOS, but i dont currently have access to a mac, so if i added the macOS dependencies i would not be confident that it would run.
FreeBSD could work fine in theory, but *BSD are not platforms that are mentioned as supported or tested.
I dont want to advertise support for platforms that are likely broken, but if them breaking isnt a blocker, setting platforms to unix would be an option.

Comment (hut) [reply pkgs/by-name/ou/outfly/package.nix:94]: I also have never tested it on FreeBSD or MacOS

Comment (71rd) [reply pkgs/by-name/ou/outfly/package.nix]: in that case i would prefer leaving it linux-only for now

**Review by zimward — APPROVED** [review]
LGTM

**Comment by 71rd** [comment]
@emilazy i tried to address  the changes you requested, it would be great if you could re-review this when you get around to it

**Comment by 71rd** [comment]
rebased and updated to new release

File: pkgs/by-name/ou/outfly/package.nix
Comment (pluiedev) [inline pkgs/by-name/ou/outfly/package.nix:70]: See pkgs/README.md
```suggestion
    description = "Breathtaking 3D space game in the rings of Jupiter";
```

File: pkgs/by-name/ou/outfly/package.nix
Comment (pluiedev) [inline pkgs/by-name/ou/outfly/package.nix:25]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/by-name/ou/outfly/package.nix
Comment (pluiedev) [inline pkgs/by-name/ou/outfly/package.nix:33]: Would having enable flags for X11 and Wayland be useful? (Non-blocking, might be future work)

**Review by pluiedev — APPROVED** [review]
Just some tiny nitpicks

File: pkgs/by-name/ou/outfly/package.nix
Comment (71rd) [reply pkgs/by-name/ou/outfly/package.nix:70]: thanks for noticing

File: pkgs/by-name/ou/outfly/package.nix
Comment (71rd) [reply pkgs/by-name/ou/outfly/package.nix:33]: This pull-request originally had feature flags for x11, wayland, openGL, and vulkan, but those were removed to simplify the package.
For this and other reasons the package was built with ''noDefaultFeatures" and required explicitly enabling features.

The current release does not require explicitly selecting features, since both game and pull-request changed since it was first submitted, and just happened to still be there

**Review by pluiedev — APPROVED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 337116`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>outfly</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>outfly</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>outfly</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>outfly</li>
  </ul>
</details>

**Pull request merged**
